tests/unit/config/test_project.py

```diff
@@ -402,6 +402,30 @@ def test_custom_query_comment_append(self):
         )
         self.assertEqual(project.query_comment.comment, "run by user test")
         self.assertEqual(project.query_comment.append, True)
+    
+    def test_default_query_comment_append_False(self):
```

Comment:
added explicit tests for when dbt-core drives the setting of "append" to False in the project